### PR TITLE
Heads of Staff can no longer roll traitor or changeling if its lowpop

### PR DESCRIPTION
**Pull request by Takahiru**
Being able to get funny roundstart all access and a free not suspicious pass because there was no captain is pretty darn powerful, it can pretty much guarantee a greentext may someone roll a head of staff as antag, and there's no captain present. This can create a problem on lowpop.

Controversial!

### CODE REVIEW

**Comment by yacabo** [comment]
What happened to https://github.com/yogstation13/Yogstation/pull/8829?

**Comment by Takahiru** [comment]
i closed it and reoponed another one

File: code/game/gamemodes/changeling/changeling.dm [hunk 1]
```diff
@@ -28,6 +28,9 @@ GLOBAL_VAR(changeling_team_objective_type) //If this is not null, we hand our th
 
 /datum/game_mode/changeling/pre_setup()
 
+	if(num_players() < 31)
```
Comment (Bibby0110) [inline code/game/gamemodes/changeling/changeling.dm]: if(num_players() <= 30)

File: code/game/gamemodes/traitor/traitor.dm [hunk 2]
```diff
@@ -33,6 +33,9 @@
 
 /datum/game_mode/traitor/pre_setup()
 
+	if(num_players() < 31)
```
Comment (Bibby0110) [inline code/game/gamemodes/traitor/traitor.dm]: if(num_players() <= 30)

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
This should probably also be a define somewhere

**Review by JamieD1 — APPROVED** [review]

**Comment by alexkar598** [comment]
I'd rather have this be a percentage based on the amounts of players tbh, to keep the oh shit he might be a baddie but without the hes a baddie every round

**Comment by Jarod1200** [comment]
Don't forget the free stroll into the armory since there's no sec and you implanted yourself

**Comment by Takahiru** [comment]
Hopefully this code works, it's hopefully supposed to follow this chart here.
image

**Comment by ynot01** [comment]
drats now how will i get antag as 4 pop CE

**Comment by Firewolf34** [comment]
Yes omg. It makes it too damn easy. The crew can't do anything against you if you just rush the spare as a head of staff in the first 5 minutes. Just pretend to be Captain... bang, all objectives complete. On Low Pop there's not even the chance anyone will try to stop you.

File: code/game/gamemodes/game_mode.dm [hunk 3]
```diff
@@ -26,6 +26,8 @@
 	var/list/restricted_jobs = list()	// Jobs it doesn't make sense to be.  I.E chaplain or AI cultist
 	var/list/protected_jobs = list()	// Jobs that can't be traitors because
 	var/list/required_jobs = list()		// alternative required job groups eg list(list(cap=1),list(hos=1,sec=2)) translates to one captain OR one hos and two secmans
+	var/list/lowpop_restricted_jobs = list() // Jobs that are restricted because the number of players are too low
```
Comment (alexkar598) [inline code/game/gamemodes/game_mode.dm]: Use GLOB.command_positions instead of yet another list

File: code/game/gamemodes/traitor/traitor.dm [hunk 4]
```diff
@@ -13,11 +13,13 @@
 	false_report_weight = 20 //Reports of traitors are pretty common.
 	restricted_jobs = list("Cyborg")//They are part of the AI if he is traitor so are they, they use to get double chances
 	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel") //YOGS -  added the hop
+	lowpop_restricted_jobs = list("Chief Engineer", "Research Director", "Chief Medical Officer")
```
Comment (alexkar598) [inline code/game/gamemodes/traitor/traitor.dm]: ~~you said command, wheres QM~~

File: code/game/gamemodes/traitor/traitor.dm [hunk 5]
```diff
@@ -33,6 +35,10 @@
 
 /datum/game_mode/traitor/pre_setup()
 
+	if(num_players() <= lowpop_amount)
+		if(prob((2*1.14^lowpop_amount)-2)) //exponential equation, chance of restriction goes up as pop goes down.
```
Comment (alexkar598) [inline code/game/gamemodes/traitor/traitor.dm]: ^ is XOR, ** is power

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/traitor/traitor.dm (hunk 5 above)
Comment (Takahiru) [reply code/game/gamemodes/traitor/traitor.dm]: yes

File: code/game/gamemodes/changeling/changeling.dm [hunk 6]
```diff
@@ -28,6 +29,10 @@ GLOBAL_VAR(changeling_team_objective_type) //If this is not null, we hand our th
 
 /datum/game_mode/changeling/pre_setup()
 
+	if(num_players() <= lowpop_amount)
+		if(prob((2*1.14**lowpop_amount)-2)) //exponential equation, chance of restriction goes up as pop goes down.
```
Comment (alexkar598) [inline code/game/gamemodes/changeling/changeling.dm]: You have the thing in the wrong thingy, the way you have it, the more players you have, the less chance you have to have a traitor head

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/changeling/changeling.dm (hunk 6 above)
Comment (jcatjmeow) [inline code/game/gamemodes/changeling/changeling.dm]:
```suggestion
		if(prob((2*1.14**num_players())-2)) //exponential equation, chance of restriction goes up as pop goes down.
```

File: code/game/gamemodes/changeling/changeling.dm [hunk 7]
```diff
@@ -17,6 +17,7 @@ GLOBAL_VAR(changeling_team_objective_type) //If this is not null, we hand our th
 	required_enemies = 1
 	recommended_enemies = 4
 	reroll_friendly = 1
+	lowpop_amount = 30
```
Comment (Bibby0110) [inline code/game/gamemodes/changeling/changeling.dm]: why are you redefining this here, it's already inherited as 30

File: code/game/gamemodes/traitor/traitor.dm [hunk 8]
```diff
@@ -18,6 +18,7 @@
 	recommended_enemies = 4
 	reroll_friendly = 1
 	enemy_minimum_age = 0
+	lowpop_amount = 30
```
Comment (Bibby0110) [inline code/game/gamemodes/traitor/traitor.dm]: see other comment

File: code/game/gamemodes/changeling/changeling.dm (hunk 6 above)
Comment (jcatjmeow) [inline code/game/gamemodes/changeling/changeling.dm]:
```suggestion
		if!(prob((2*1.14**num_players())-2)) //exponential equation, chance of restriction goes up as pop goes down.
```

**Review by alexkar598 — APPROVED** [review]

**Pull request merged**